.github/workflows/precommit.yaml

```diff
@@ -1,4 +1,5 @@
 name: pre-commit
+
```

Comment:
Should probably just call this workflow "Linting" or "Code Quality" or something.  Yes we're using `pre-commit` but that's more of a runner vs. being descriptive of what checks are actually being performed here.

Reply:
Done in https://github.com/mosaicml/streaming/pull/50/commits/052061bd11fb8c286bb726c73bf10fec330e1398.

Reply:
We should rename this file as well to match the workflow name.

Reply:
Done in https://github.com/mosaicml/streaming/pull/50/commits/796fbdbbbf7cad3179d8651314f4f7f171d10ac0.